packages/react-menu/src/components/MenuPopover/useMenuPopover.ts

```diff
@@ -39,31 +38,31 @@ export const useMenuPopover = (
     props,
   );
 
-  state.isSubmenu = useMenuContext(context => context.isSubmenu);
   state.inline = useMenuContext(context => context.inline);
 
-  const { onMouseEnter: onMouseEnterOriginal, onBlur: onBlurOriginal, onKeyDown: onKeyDownOriginal } = state;
+  const { onMouseEnter: onMouseEnterOriginal, onKeyDown: onKeyDownOriginal } = state;
 
   state.onMouseEnter = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (popoverRef.current) {
+      dispatchMenuEnterEvent(popoverRef.current);
+    }
     if (openOnHover && !openOnContext) {
       setOpen(e, { open: true, keyboard: false });
     }
 
     onMouseEnterOriginal?.(e);
   });
 
-  state.onBlur = useEventCallback((e: React.FocusEvent<HTMLElement>) => {
-    if (isOutsideMenu({ triggerRef, menuPopoverRef: popoverRef, event: e })) {
-      setOpen(e, { open: false, keyboard: false });
+  state.onKeyDown = useEventCallback((e: React.KeyboardEvent<HTMLElement>) => {
+    if (e.key === 'Escape' || (isSubmenu && e.key === 'ArrowLeft')) {
```

Comment:
Wouldn't make more sense to use `@fluentui/keyboard-key` for key code controlling?

Reply:
No, we no longer support IE11 and the entire library is un-treeshakeable... upcoming RFC soon about that. other packages like react-tooltip already don't use it